Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -11236,6 +11316,57 @@ - (void)dropInteraction:(UIDropInteraction *)interaction sessionDidEnd:(id<UIDro
 
 #endif // ENABLE(DRAG_SUPPORT)
 
+#pragma mark - Pan Gesture Support
+
+-(void)panGestureDidBeginAtPoint:(CGPoint)inputPoint {
+#if ENABLE(MODEL_PROCESS)
+    if (_stageModeSession.state == WebKit::StageModeSessionState::Idle) {
+        [self cleanUpStageModeSessionState];
+        _page->requestInteractiveModelElementAtPoint(WebCore::roundedIntPoint(inputPoint));
+        _stageModeSession.state = WebKit::StageModeSessionState::Preparing;
+    }
+#endif // ENABLE(MODEL_PROCESS)
+}
+
+// FIXME: https://bugs.webkit.org/show_bug.cgi?id=287601
+-(void)panGestureDidUpdateWithPoint:(CGPoint)inputPoint {
+#if ENABLE(MODEL_PROCESS)
+    if (_stageModeSession.state == WebKit::StageModeSessionState::Active) {
+        [self updateStageModeSessionTransformWithPoint:inputPoint];
+        _page->stageModeSessionDidUpdate(_stageModeSession.elementID, _stageModeSession.transform);
+    }
+#endif // ENABLE(MODEL_PROCESS)
+}
+
+-(void)panGestureDidEnd {
+#if ENABLE(MODEL_PROCESS)
+    if (_stageModeSession.state == WebKit::StageModeSessionState::Active)
+        _page->stageModeSessionDidEnd(_stageModeSession.elementID);
+
+    [self cleanUpStageModeSessionState];
+#endif // ENABLE(MODEL_PROCESS)
+}
+
+#if ENABLE(MODEL_PROCESS)
+-(void)didReceiveInteractiveModelElement:(std::optional<WebCore::ElementIdentifier>)elementID {
+    if (_stageModeSession.state != WebKit::StageModeSessionState::Preparing)
+        return;
+    _stageModeSession.state = elementID ? WebKit::StageModeSessionState::Active : WebKit::StageModeSessionState::Idle;
+    _stageModeSession.elementID = elementID;
+}
+
+-(void)cleanUpStageModeSessionState {
+    _stageModeSession = { };
+}
+
+-(void)updateStageModeSessionTransformWithPoint:(CGPoint)inputPoint {
+    WebCore::TransformationMatrix tf;
+    tf.makeIdentity();
```

Comment:
I think the default ctor already creates the identity matrix.

Also, `tf` should be `transform` or `matrix` (we prefer not to use letter abbreviations for variable names in WebKit code)